### PR TITLE
Add support for Cynosa Lite

### PR DESCRIPTION
**Pull request by oviradoi**
Adds support for the Cynosa Lite keyboard.

The following effects work:
 - matrix_effect_none
 - matrix_effect_static
 - matrix_effect_breath
 - matrix_effect_spectrum

The following effects do not work:
 - matrix_effect_wave
 - matrix_effect_reactive

This addresses issue #1027

### CODE REVIEW

**Comment by z3ntu** [comment]
Hi, thanks for the PR! There are a couple of things missing though for the PR to be merged.

1. Add an entry for your device to `README.md`
2. Add a definition for the device in the daemon (`daemon/openrazer_daemon/hardware/keyboards.py`)
3. Run `./scripts/generate_appstream_file.sh` to regenerate `install_files/appstream/io.github.openrazer.openrazer.metainfo.xml`
4. Update `install_files/udev/99-razer.rules` and include the PID of your device in the keyboards section

**Comment by oviradoi** [comment]
Thanks for reviewing the PR.

I have added the missing stuff, but I'm not really sure if the python file is correct, since I'm not familiar with it.

**Comment by z3ntu** [comment]
You can easily test the whole thing by installing a frontend like polychromatic or RazerGenie, seeing if all lighting options are available there and clicking around to see if the effects actually work.

**Comment by oviradoi** [comment]
~~@z3ntu I think I need some help with this. The driver works fine for my keyboard when I echo stuff into the files in /sys/bus/hid/drivers/razerkbd/. However, if I install Polychromatic or RazerGenie, they don't detect my keyboard.~~

~~I have also tried using d-feet to get the list of devices from the daemon, and it returns an empty list. If I get the list of supported devices, the Cynosa Lite is not in the list. Do you know what I'm doing wrong?~~

Nevermind, this started to work after I unplugged my keyboard and plugged it in again. Polychromatic and RazerGenie work with my keyboard, and I can use it to set the different color modes.

File: README.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ The below devices are fully feature supported by OpenRazer which means all avail
 | Razer Huntsman                                |  1532:0227  |
 | Razer BlackWidow Elite                        |  1532:0228  |
 | Razer Cynosa Chroma                           |  1532:022A  |
+| Razer Cynosa Lite                             |  1532:023F  |
```
Comment (z3ntu) [inline README.md]: Please put this entry after `1532:023B` so it's sorted by usb id

File: daemon/openrazer_daemon/hardware/keyboards.py [hunk 2]
```diff
@@ -873,6 +873,26 @@ class RazerCynosaChroma(_RippleKeyboard):
...
+    USB_PID = 0x023F
+    METHODS = ['get_device_type_keyboard', 'set_static_effect', 'set_spectrum_effect',
+               'set_none_effect', 'set_breath_random_effect', 'set_breath_single_effect', 'set_breath_dual_effect',
+               'get_game_mode', 'set_game_mode']
+
+    DEVICE_IMAGE = "https://www.windowscentral.com/sites/wpcentral.com/files/styles/w830_wm_brw/public/field/image/2020/01/razer-cynosa-lite-review-1.jpg?itok=HDtoNRR2"
+
+    # Deprecated - RAZER_URLS be removed in future.
+    RAZER_URLS = {
+        "top_img": "https://www.windowscentral.com/sites/wpcentral.com/files/styles/w830_wm_brw/public/field/image/2020/01/razer-cynosa-lite-review-1.jpg?itok=HDtoNRR2"
```
Comment (z3ntu) [inline daemon/openrazer_daemon/hardware/keyboards.py]: I'm not happy with linking to some third-party site. I can't seem to find the product on the Razer website but maybe you can find an image there which looks similar enough to the Cynosa Lite?

File: driver/razerkbd_driver.h [hunk 3]
```diff
@@ -46,6 +46,7 @@
 #define USB_DEVICE_ID_RAZER_HUNTSMAN 0x0227
 #define USB_DEVICE_ID_RAZER_BLACKWIDOW_ELITE 0x0228
 #define USB_DEVICE_ID_RAZER_CYNOSA_CHROMA 0x022A
+#define USB_DEVICE_ID_RAZER_CYNOSA_LITE 0x023F
```
Comment (z3ntu) [inline driver/razerkbd_driver.h]: put it below USB_DEVICE_ID_RAZER_BLADE_2018_BASE please

File: install_files/udev/99-razer.rules [hunk 4]
```diff
@@ -10,7 +10,7 @@ ATTRS{idProduct}=="0013|0016|0020|0024|0025|002e|002f|0032|0034|0036|0037|0038|0
     ENV{ID_RAZER_CHROMA}="1", ENV{RAZER_DRIVER}="razermouse"
 
 # Keyboards
-ATTRS{idProduct}=="010d|010e|010f|0111|0113|011a|011b|0201|0202|0203|0204|0205|0207|0208|0209|020f|0210|0211|0214|0216|0217|021a|021e|021f|0220|0221|0224|0225|0226|0227|0228|022a|022b|022d|022f|0232|0233|0234|0235|0237|0239|023a|023b|0240|0241|0243|0245|0246|024a|024c|024d", \
+ATTRS{idProduct}=="010d|010e|010f|0111|0113|011a|011b|0201|0202|0203|0204|0205|0207|0208|0209|020f|0210|0211|0214|0216|0217|021a|021e|021f|0220|0221|0224|0225|0226|0227|0228|022a|023f|022b|022d|022f|0232|0233|0234|0235|0237|0239|023a|023b|0240|0241|0243|0245|0246|024a|024c|024d", \
```
Comment (z3ntu) [inline install_files/udev/99-razer.rules]: also sort by pid here

**Comment by oviradoi** [comment]
Thanks for the review, I pushed the modifications to the branch.

I'm not sure why this keyboard does not show up on Razer's website, it must not have been added yet. I used the photo url from Cynosa Chroma. They look the same, but the Lite version only supports single-zone lighting.

**Comment by oviradoi** [comment]
I had some more time to test some features, and it looks like the macro recording also works. I have enabled macro support in this pull request too.

**Comment by andreimarinescu** [comment]
@oviradoi is there any reasonably quick way of building and installing your branch while waiting for merge? I'm really curious to test this out! (sorry for posting this here, didn't find build instructions in README.md and wasn't sure how to proceed).

**Comment by oviradoi** [comment]
@andreimarinescu You can use the instructions in the wiki here: https://github.com/openrazer/openrazer/wiki/Building-a-package

In short, for Ubuntu, you need to run `build_debs.sh` to build the packages and then install them using `dpkg`.
You should uninstall the stable version first, because it might cause problems since both of them will have the same version number.

**Comment by andreimarinescu** [comment]
Thanks! I've seen the wiki page as well, I'm running Arch though so the instructions where just regarding building the package from AUR. 

Since I've never built a KMS module before so I thought it's worth asking before I nuke my install. I don't want to pollute this thread further, I'll try to figure things out. 

Looking forward to seeing this released, and thanks for contributing this, from a future daily user!

**Comment by andreimarinescu** [comment]
@oviradoi been using this for the better part of this week. Works well, even when paired with another DeathAdder mouse. Only issue - are custom colors actually supported? I can't seem to be able to select one (using polychromatic-controller in my case). Selecting a custom color doesn't seem to affect the device.

**Comment by oviradoi** [comment]
@andreimarinescu Yes, choosing a custom color in static mode in polychromatic works fine for me. Can you try sending the command directly to the OpenRazer driver, using the instructions in the wiki? https://github.com/openrazer/openrazer/wiki/Using-the-keyboard-driver#matrix_effect_static---static-effect-mode-write-only

**Comment by andreimarinescu** [comment]
Thanks, @oviradoi. Indeed, it does work when setting it this way. After further testing, it seems that at least for me, I need to select another lighting effect and go back to the static setting, when using polychromatic in order for the setting to be applied. But that would be a polychromatic issue, maybe it has to do with my WM. Thanks again!

**Comment by z3ntu** [comment]
Is this PR good to go from your side?

**Comment by oviradoi** [comment]
Yes, the PR is good to go from my side, but I assume you wanted to ask @andreimarinescu

**Comment by andreimarinescu** [comment]
Sorry guys, I assumed you wanted to ask @oviradoi. Yes, this PR has been working stable on my machine for the past 7-8 days (Manjaro w/ 5.5.19 kernel, Razer Cynosa Lite + DeathAdder Essential).

**Comment by oviradoi** [comment]
I have updated the pull request with the proper image for the Razer Cynosa Lite. It looks like they finally added it to their website.

Is there anything else I need to do in order to get this merged?

**Review by z3ntu — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**